cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTimePartitionedFileSet.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch;
+
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.FileSetArguments;
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.api.dataset.lib.TimePartitionedFileSet;
+import co.cask.cdap.api.dataset.lib.TimePartitionedFileSetArguments;
+import co.cask.cdap.api.dataset.table.Put;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.TextOutputFormat;
+
+import java.io.IOException;
+
+/**
+ * App used to test whether M/R works well with time-partitioned file sets.
+ * It uses M/R to read from a table and write partitions, and another M/R to read partitions and write to a table.
+ */
+public class AppWithTimePartitionedFileSet extends AbstractApplication {
+
+  public static final String INPUT = "input";
+  public static final String PARTITIONED = "partitioned";
+  public static final String OUTPUT = "output";
+  public static final byte[] ONLY_COLUMN = { 'x' };
+  public static final String ROW_TO_WRITE = "row.to.write";
+  private static final String SEPARATOR = ":";
+
+  @Override
+  public void configure() {
+    try {
+      setName("AppWithMapReduceUsingFile");
+      setDescription("Application with MapReduce job using file as dataset");
+      createDataset(INPUT, "table");
+      createDataset(OUTPUT, "table");
+
+      createDataset(PARTITIONED, "timePartitionedFileSet", FileSetProperties.builder()
+        // properties for file set
+        .setBasePath("/partitioned")
+        .setInputFormat(TextInputFormat.class)
+        .setOutputFormat(TextOutputFormat.class)
+        .setOutputProperty(TextOutputFormat.SEPERATOR, SEPARATOR)
+          // properties for partitioned hive table
+          // .setEnableExploreOnCreate(true)
```

Comment:
remove commented properties unless we're adding them back in another PR very soon.